ui/app/components/transaction-list-item/transaction-list-item.component.js

```diff
@@ -39,31 +37,16 @@ export default class TransactionListItem extends PureComponent {
     const {
       transaction,
       history,
-      showTransactionDetailsModal,
-      methodData,
-      showCancel,
-      showRetry,
     } = this.props
     const { id, status } = transaction
     const { showTransactionDetails } = this.state
-    const windowType = window.METAMASK_UI_TYPE
 
     if (status === UNAPPROVED_STATUS) {
       history.push(`${CONFIRM_TRANSACTION_ROUTE}/${id}`)
       return
     }
 
-    if (windowType === ENVIRONMENT_TYPE_FULLSCREEN) {
-      this.setState({ showTransactionDetails: !showTransactionDetails })
-    } else {
-      showTransactionDetailsModal({
-        transaction,
-        onRetry: this.handleRetry,
-        showRetry: showRetry && methodData.done,
-        onCancel: this.handleCancel,
-        showCancel,
-      })
-    }
+    this.setState({ showTransactionDetails: !showTransactionDetails })
```

Comment:
We should be using the version of `setState` that accepts a function of the previous state